### ISSUE TITLE
wrong separator in getcomputedstyle or .css() (locale issue?)

### ISSUE THREAD

**Issue by bobatsar** [nwjs/nw.js#4790]
I am using nw.js 0.14.2 on osx 10.9.5 and have issues when computing styles from the dom.

It seems the locale is used to calculate the values. (in german a comma is used instead of a period)

Example package.json:

```
{
"name": "testapp",
"main": "http://www.heise.de",
}
```

In the developer console:

```
$(".heise_foren").css("font-size")
"12,8px"
```

The result should be "12.8px" and not "12,8px" and can of course not be used for further processing.
The same issue arises when getComputedStyle() is used.

The locale should not be used for such calculations.

On Chrome Version 50.0.2661.86 (64-bit) in OSX it is working as expected:

```
getComputedStyle($(".heise_foren")).getPropertyValue("font-size")
"12.8px"
```

**Comment by rogerwang** [comment]
I tried latest 0.14.4 under German locale and it works for me. See attached screenshot. Could you please check that version?

![screen shot 2016-05-09 at 13 12 39](https://cloud.githubusercontent.com/assets/165401/15104256/1cd0ffaa-15e8-11e6-8436-db8198a49b74.png)

Please reopen if it still doesn't work for you.

**Comment by bobatsar** [comment]
@rogerwang, 
I can't reopen, can you please reopen it?

I just tried with nw 0.14.4 and the problem is still there but it seems the problem is not the locale as navigator.language is "en-US".
On OSX 10.9.5 the following happens, but on linux and windows it seems ok.

See the following gist for the index.html I use.
https://gist.github.com/9e24797324b7f2a4ca46ec2ab5028390

![bildschirmfoto 2016-05-09 um 09 46 36](https://cloud.githubusercontent.com/assets/7723101/15106656/263c016a-15cb-11e6-93a1-30468caf7f9a.png)

Do you need some more debugging information?

**Comment by bobatsar** [comment]
Just wanted to add that this still happens with nw 0.20.3

I realised that if you change the language in OSX and restart the test, the displayed navigator.language is correctly updated to the new language, but the size string stays in the old locale until you restart.
If I change the language to "English", do a restart, then the size is correctly displayed with a period. If I change back to german and restart the system, the problem reappears.

The string from the CSSStyleDeclaration Object (returned from getComputedStyle() ) shows already that wrong result.

It looks like the string is calculated somehow based on the locale.

**Comment by rogerwang** [comment]
cc @Christywl

**Comment by Christywl** [comment]
I checked on Mac with nwjs-sdk-0.20.3, didn't reproduce. 
language is 'English':
![2](https://cloud.githubusercontent.com/assets/20039821/23738851/4d9d7482-04d6-11e7-87e6-bd27a60b7df1.png)
language is 'German':
![1](https://cloud.githubusercontent.com/assets/20039821/23738848/4c534ee4-04d6-11e7-809e-2086ce547c81.png)

**Comment by bobatsar** [comment]
Did you restart your system after switching to German? I see the same behaviour if I only switch the language and then just restart the test. Then the language is switched but the output stays the same.

If I do the same with Chromium everything is OK.
![bildschirmfoto 2017-03-09 um 09 00 03](https://cloud.githubusercontent.com/assets/7723101/23741112/34e6dca0-04a7-11e7-9eff-8f2d6e9f35cf.png)

@Christywl How do you switch the language? Just in the OSX settings or do you do something different?
I am still on OSX 10.9.5. Could this be the problem?

**Comment by Christywl** [comment]
I changed the language in 'System Preferences/Language & Region':
1. Changed the language to 'German' and restart the test, the navigator.language is 'de', and the size is '21.5px'(notes: the system language is still English because I did not restart the Mac)
2. And then restart the Mac, and  the system language is changed to 'German'. Start the test again, the navigator.language is 'de' and the size is '21.5px'.

My Mac is 10.11.6.

**Comment by Christywl** [comment]
@bobatsar , could you please check your settings as below?
`System Preferences-->Language & Region-->Region`

If it sets to 'Germany', the result is same as yours. 'Region' sets 'United States(Computer)'. After changed the Region, please restart the PC.

**Comment by bobatsar** [comment]
Yes it is as you describe. If I let the language in German but set the region to United States, the output is correct. But if I have language and region set to German, than the output is wrong.

**Comment by Christywl** [comment]
I think we could close this now.

**Comment by bobatsar** [comment]
@Christywl why do you close this issue? Do you need more information? 
The error still exists and it is reproducible when the region is set to germany.

Chrome 57 does not have this issue but node-webkit 0.21.6 still has it.

**Comment by sourceskyboxer** [comment]
Hello my dears,

If they don't fix nw.js locale than I suggest with other method:
[Link here](http://stackoverflow.com/questions/17493309/how-do-i-change-the-language-of-moment-js)
Is it possible to use Language Chooser and Preference saves to ApplicationStorageDirectoy -> nw-app/pref.up ( just save custom preference file and you don't need worry - If your nw-app reloads current saved locale. ) Sorry for longer than month!
